src/webhook/pkg/mutating_webhook/remove-resourcespec.go

```diff
@@ -26,18 +35,96 @@ func (a *PodMutator) mutatePod(pod *corev1.Pod) (*corev1.Pod, error) {
 		return mpod, nil
 	}
 
+	mpod = adjustResourceSpec(mpod)
+
+	return mpod, nil
+}
+
+// function to remove resource spec from the pod spec
+// add the cumulative resources as annotation to pod spec
+// add the peer-pod resource to the first container in the pod spec
+
+func adjustResourceSpec(pod *corev1.Pod) *corev1.Pod {
+
+	// Get total CPU resource requests
+	cpuRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceCPU)
+
+	// Get total CPU resource limits
+	cpuLimit := utils.GetResourceLimitQuantity(pod, corev1.ResourceCPU)
+
+	// Get total Memory resource requests
+	memoryRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceMemory)
+
+	// Get total Memory resource limits
+	memoryLimit := utils.GetResourceLimitQuantity(pod, corev1.ResourceMemory)
+
+	// Get total GPU resource requests
+	// GPU resources are always requested in whole numbers and request and limits are same.
+	// So we don't need to check for limits
+	gpuRequest := utils.GetResourceRequestQuantity(pod, corev1.ResourceName(GPU_RESOURCE_NAME))
+
+	// log the resource values
+	logger.Printf("CPU Request: %s, CPU Limit: %s, Memory Request: %s, Memory Limit: %s, GPU Request: %s",
+		cpuRequest.String(), cpuLimit.String(), memoryRequest.String(), memoryLimit.String(), gpuRequest.String())
+
+	// Add the cumulative resources as annotation to pod spec
+	// Use cpuLimit and memoryLimit if those are greater than cpuResource and memoryResource
+	annotations := pod.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	// A non-existent request or limit will be 0 and we don't need to add annotation for 0
+	// We only add annotation if the value is greater than 0
+	// Limit will always be preferred over request
+
+	// Add cpu annotation
+	if !cpuRequest.IsZero() && cpuLimit.Cmp(cpuRequest) >= 0 {
+		logger.Printf("Adding CPU annotation based on cpuLimit: %s", cpuLimit.String())
+		annotations[PEERPODS_CPU_ANNOTATION] = cpuLimit.String()
+	} else if cpuRequest.Sign() == 1 {
+		logger.Printf("Adding CPU annotation based on cpuRequest: %s", cpuRequest.String())
+		annotations[PEERPODS_CPU_ANNOTATION] = cpuRequest.String()
+	}
+
+	// Add memory annotation
+	if !memoryRequest.IsZero() && memoryLimit.Cmp(memoryRequest) >= 0 {
+		logger.Printf("Adding Memory annotation based on memoryLimit: %s", memoryLimit.String())
+		memoryLimitMiBStr, err := utils.ConvertMemoryQuantityToMib(memoryLimit)
+		if err != nil {
+			logger.Printf("Error converting memory quantity to MiB: %v", err)
+		}
+		annotations[PEERPODS_MEMORY_ANNOTATION] = memoryLimitMiBStr
```

Comment:
Hi @bpradipt !

Should it really set annotations[PEERPODS_MEMORY_ANNOTATION] if conversion to memoryLimitMiBStr failed?

Reply:
Yeah, it'll be 0 and the error is logged to see why it failed. It's not a strict failure..

Reply:
Let me know if you want me to add a comment to make it explicit. Although looking at the `ConvertMemoryQuantityToMib` method will make it clear that it returns "0" and the same is set as the annotation.

Reply:
ok. I didn't look at `ConvertMemoryQuantityToMib` and I was worried it returning some invalid value. It's okay as is